src/libraries/System.Private.Xml.Linq/tests/misc/LoadSaveAsyncTests.cs

```diff
@@ -218,53 +218,184 @@ public async Task SaveAsync_CallsAsyncOnly_SaveSync_CallsSyncOnly(bool isAsync,
                     await element.SaveAsync(stream, saveOptions, CancellationToken.None);
                 }
-                else
+            }
+            else
+            {
+                using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
                 {
                     document.Save(stream);
                     element.Save(stream);
                 }
             }
         }
+
+        [Theory]
+        [MemberData(nameof(RoundtripSyncAsyncOnly_MemberData))]
+        public static async Task RoundtripSyncAsyncOnly(bool async, bool document, LoadOptions loadOptions, SaveOptions saveOptions)
+        {
+            ForceSyncAsyncStream outputStream = new ForceSyncAsyncStream(async);
+            
+            if (async)
+            {
+                await using (Stream inputStream = new ForceSyncAsyncStream(async, FilePathUtil.getStream(GetTestFileName())))
+                {
+                    if (document)
+                    {
+                        XDocument asyncDoc = await XDocument.LoadAsync(inputStream, loadOptions, CancellationToken.None);
+                        await asyncDoc.SaveAsync(outputStream, saveOptions, CancellationToken.None);
+                    }
+                    else
+                    {
+                        XElement asyncElement = await XElement.LoadAsync(inputStream, loadOptions, CancellationToken.None);
+                        await asyncElement.SaveAsync(outputStream, saveOptions, CancellationToken.None);
+                    }
+                }
+            }
+            else
+            {
+                using (Stream inputStream = new ForceSyncAsyncStream(async, FilePathUtil.getStream(GetTestFileName())))
+                {
+                    if (document)
+                    {
+                        XDocument syncDoc = XDocument.Load(inputStream, loadOptions);
+                        syncDoc.Save(outputStream, saveOptions);
+                    }
+                    else
+                    {
+                        XElement syncElement = XElement.Load(inputStream, loadOptions);
+                        syncElement.Save(outputStream, saveOptions);
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> RoundtripSyncAsyncOnly_MemberData
+        {
+            get
+            {
+                foreach (bool async in new[] { false, true })
+                    foreach (bool doc in new[] { true, false })
+                        foreach (LoadOptions loadOptions in Enum.GetValues(typeof(LoadOptions)))
+                            foreach (SaveOptions saveOptions in Enum.GetValues(typeof(SaveOptions)))
+                                yield return new object[] { async, doc, loadOptions, saveOptions };
+            }
+        }
     }
 
-    public class ForceSyncAsyncStream : MemoryStream
+    public class ForceSyncAsyncStream : Stream
```

Comment:
Could you consider creating a separate testing stream? I see the point you changing it, but this stream was dedicated for checking only async method called for async call so that async flush would never call sync write, same for async dispose

Reply:
OK. I created a separate testing stream.

Reply:
Thanks!